web/src/rendered_markdown.ts

```diff
@@ -391,5 +393,34 @@ export const update_elements = ($content: JQuery): void => {
             })
             .contents()
             .unwrap();
+    } else if (
+        user_settings.web_animate_image_previews !==
+        settings_config.web_animate_image_previews_values.always.code
+    ) {
+        $content.find("img.emoji").each(function (): void {
+            const $emoji = $(this);
+            const emoji_url = $emoji.attr("src") ?? "";
+            const emoji_obj = emoji.all_realm_emojis_by_url.get(emoji_url);
+            if (
+                emoji_obj?.still_url &&
+                user_settings.web_animate_image_previews ===
+                    settings_config.web_animate_image_previews_values.on_hover.code
+            ) {
+                const $still_emoji = $("<img>");
+
+                $still_emoji.addClass("emoji still-emoji");
+                $still_emoji.attr("src", emoji_obj.still_url);
+                $still_emoji.attr("alt", `:${emoji_obj.emoji_name}:`);
+                $still_emoji.attr("title", emoji_obj.emoji_name);
+                $emoji.addClass("animated-emoji");
+                $emoji.after($still_emoji);
```

Comment:
Since this duplicates the emoji elements, I think we'll need to take some care to avoid copy/paste bringing along multiple copies of the emoji.

Reading this, I wonder a bit if it'd be better to have the emoji animate in a (enlarged) tooltip rather than having it animate in-place for the on-hover behavior? Not sure, but that seems like a direction worth exploring.